.travis.yml

```diff
@@ -1,10 +1,46 @@
-services:
-- docker
-script:
-- docker build --target test --tag todo-app:test .
-- docker run --mount type=bind,source="$(pwd)"/,target=/app/ todo-app:test tests/test_view_model.py
-- docker run --mount type=bind,source="$(pwd)"/,target=/app/ todo-app:test tests/test_endpoints.py
-- docker run  -eT_KEY=$T_KEY -eT_TOKEN=$T_TOKEN --mount type=bind,source="$(pwd)"/,target=/app/ todo-app:test tests_e2e
+services: docker
+jobs:
+ include:
+  - stage: Build docker Image
+    script: 
+    - docker build --target test --tag virendras19/todo-app:test .
+    - docker images
+    - docker login -u $DOCKER_USERNAME -p $DOCKER_CODE
+    - docker image push virendras19/todo-app:test
+
+  - stage: Test
+    before_script: docker pull virendras19/todo-app:test
+    script: docker run --mount type=bind,source="$(pwd)"/,target=/app/ virendras19/todo-app:test tests/test_view_model.py
+    name: "Unit Test"
+
+   
+  - before_script: docker pull virendras19/todo-app:test 
+    script: docker run --mount type=bind,source="$(pwd)"/,target=/app/ virendras19/todo-app:test tests/test_endpoints.py
+    name: "Endpoint Test"
+    
+  - before_script: docker pull virendras19/todo-app:test 
+    script: docker run  -eT_KEY=$T_KEY -eT_TOKEN=$T_TOKEN --mount type=bind,source="$(pwd)"/,target=/app/ virendras19/todo-app:test tests_e2e
+    name: "End to End Tests"
+
+  - stage: Deploy
+    before_script:
+    - docker login -u $DOCKER_USERNAME -p $DOCKER_CODE
+    - docker login --username=virendrasankpal@yahoo.com --password=$(heroku auth:token) registry.heroku.com
+    - heroku container:login
+    script:
+    - docker build --target production --tag virendras19/todo-app:latest --tag virendras19/todo_app:$TRAVIS_COMMIT .
+    - docker ps -a
+    - docker image list
+    - docker image push virendras19/todo-app:latest
+    - docker pull virendras19/todo-app
+    - docker tag virendras19/todo-app registry.heroku.com/virendra-todo-app/web
+    - docker push registry.heroku.com/virendra-todo-app/web
+    - heroku container:release web --app virendra-todo-app
+    - heroku open  --app virendra-todo-app
```

Comment:
This line is meaningless in the CI pipeline - it's to open the app in your browser